src/uucore/src/lib/mods/clap_localization.rs

```diff
@@ -74,316 +38,542 @@ impl Color {
-                handle_invalid_value_error(err, maybe_colorize);
-            }
+            exit_code // Respect the requested exit code for special cases
+        };
+        callback();
+        std::process::exit(actual_exit_code);
+    }
 
-            // For other simple validation errors, use the same simple format as other errors
-            if let Some(main_error_line) = rendered_str.lines().next() {
-                // Keep the "error: " prefix for test compatibility
-                eprintln!("{main_error_line}");
+    /// Handle missing required argument errors with callback
+    fn handle_missing_required_with_callback<F>(
+        &self,
+        err: &Error,
+        exit_code: i32,
+        callback: F,
+    ) -> !
+    where
+        F: FnOnce(),
+    {
+        let rendered_str = err.render().to_string();
+        let lines: Vec<&str> = rendered_str.lines().collect();
+
+        if let Some(first_line) = lines.first() {
+            if first_line.starts_with("error: the following required arguments were not provided:")
+            {
+                let error_word = translate!("common-error");
+                eprintln!(
+                    "{}",
+                    translate!(
+                        "clap-error-missing-required-arguments",
+                        "error_word" => self.color_mgr.colorize(&error_word, Color::Red)
+                    )
+                );
+
+                // Print the missing arguments
+                for line in lines.iter().skip(1) {
+                    if line.starts_with("  ") {
+                        eprintln!("{line}");
+                    } else if line.starts_with("Usage:") || line.starts_with("For more information")
+                    {
+                        break;
+                    }
+                }
                 eprintln!();
-                // Use the execution phrase for the help suggestion to match test expectations
-                eprintln!("{}", translate!("common-help-suggestion"));
-            } else {
-                // Fallback to original rendering if we can't parse
-                eprint!("{}", err.render());
-            }
 
-            // InvalidValue errors should exit with code 1 for all utilities
-            let actual_exit_code = if matches!(kind, ErrorKind::InvalidValue) {
-                1
+                // Print usage
+                let mut in_usage = false;
+                for line in &lines {
+                    if line.starts_with("Usage:") {
+                        in_usage = true;
+                    }
+                    if in_usage {
+                        if line.starts_with("For more information, try '--help'.") {
+                            eprintln!("{}", translate!("common-help-suggestion"));
+                        } else {
+                            eprintln!("{line}");
+                        }
+                    }
+                }
```

Comment:
I think a functional approach is cleaner:
```suggestion
                lines
                    .iter()
                    .skip_while(|line| !line.starts_with("Usage:"))
                    .for_each(|line| {
                        if line.starts_with("For more information, try '--help'.") {
                            eprintln!("{}", translate!("common-help-suggestion"));
                        } else {
                            eprintln!("{line}");
                        }
                    });
```